src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1503 @@ const ActorInit En_Elf_InitVars = {
+    EnElf* this = THIS;
+
+    func_80A04DE4(this, globalCtx);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && func_80106BC8(globalCtx)) {
+        func_8010B720(globalCtx, 0xE3);
+        this->actor.update = func_80A05040;
+    }
+
+    func_80A04F94(this, globalCtx);
+}
+
+void func_80A05188(Actor* thisx, GlobalContext* globalCtx) {
+    EnElf* this = THIS;
+
+    func_80A04DE4(this, globalCtx);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && func_80106BC8(globalCtx)) {
+        func_8010B720(globalCtx, ElfMessage_GetSariaText(globalCtx));
+        this->actor.update = func_80A05114;
+    }
+
+    func_80A04F94(this, globalCtx);
+}
+
+// ask to talk to navi
+void func_80A05208(Actor* thisx, GlobalContext* globalCtx) {
+    s32 naviCUpText;
+    EnElf* this = THIS;
+
+    func_80A04DE4(this, globalCtx);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && func_80106BC8(globalCtx)) {
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0: // yes
+                naviCUpText = ElfMessage_GetCUpText(globalCtx);
+
+                if (naviCUpText != 0) {
+                    func_8010B720(globalCtx, naviCUpText);
+                } else {
+                    func_8010B720(globalCtx, 0x15F);
+                }
+
+                this->actor.update = func_80A052F4;
+                break;
+            case 1: // no
+                func_80106CCC(globalCtx);
+                this->actor.update = func_80A053F0;
+                func_80A01C38(this, 0);
+                this->fairyFlags &= ~0x20;
+                break;
+        }
+    }
+
+    func_80A04F94(this, globalCtx);
+}
+
+// ask to talk to saria
+void func_80A052F4(Actor* thisx, GlobalContext* globalCtx) {
+    EnElf* this = THIS;
+
+    func_80A04DE4(this, globalCtx);
+
+    if (func_8010BDBC(&globalCtx->msgCtx) == 4) {
+        if (func_80106BC8(globalCtx)) {
+            globalCtx->msgCtx.unk_E3F2 = 0xFF;
+
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0: // yes
+                    this->actor.update = func_80A05188;
+                    func_8010B720(globalCtx, 0xE2);
+                    break;
+                case 1: // no
+                    this->actor.update = func_80A05208;
+                    func_8010B720(globalCtx, 0xE1);
+                    break;
+            }
+        }
+    } else if (func_8002F334(thisx, globalCtx)) {
+        this->actor.update = func_80A053F0;
+        func_80A01C38(this, 0);
+        this->fairyFlags &= ~0x20;
+    }
+
+    func_80A04F94(this, globalCtx);
+}
+
+void func_80A053F0(Actor* thisx, GlobalContext* globalCtx) {
+    u8 unk2C7;
+    s32 pad;
+    Player* player = PLAYER;
+    EnElf* this = THIS;
+
+    if (player->naviMessageId == 0) {
+        if (player->unk_664 == NULL) {
+            if (((gSaveContext.naviTimer >= 600) && (gSaveContext.naviTimer <= 3000)) || (nREG(89) != 0)) {
+                player->naviMessageId = ElfMessage_GetCUpText(globalCtx);
```

Comment:
We should probably rename this field since we use the term "text id", not "message" like they did originally
```suggestion
                player->naviTextId = ElfMessage_GetCUpText(globalCtx);
```